eclipse-collections-code-generator/src/main/resources/api/block/function/primitiveFunction.stg

```diff
@@ -22,7 +22,7 @@ import java.io.Serializable;
  *
  * @since 3.0.
  */
-<if(primitive.specializedStream)>@FunctionalInterface<endif>
+@FunctionalInterface
```

Comment:
@motlin I tried what you suggested. The only inspection warning I got was about replacing the fully qualified name with import in `DoubleFunction`. I tried `CharFunction` and no inspection warnings were thrown. Is there an inspection setting I am missing?
May be to begin with the prior `if` condition was incorrect in the template file. Looking at the code:
```
public boolean hasSpecializedStream()
{
    return this == INT || this == LONG || this == DOUBLE;
}
```
That means `DoubleFunction` will get the `@FunctionalInterface` annotation, where as `CharFunction` will not get that annotation. It should be the other way around, since `CharFunction` does not inherit the annotation from any other interface.
IMO we should keep the redundant annotation. If IntelliJ shows the warning, we can suppress it accordingly in the templates 
`<if(primitive.specializedStream)>//Suppress Redundant Annotation (or whichever suppression it is once the warning comes up to figure it out)<endif>`